benchmarks/distributed/rpc/ps/metrics/CUDAMetric.py

```diff
@@ -0,0 +1,25 @@
+import torch
+
+from .MetricBase import MetricBase
+
+
+class CUDAMetric(MetricBase):
+    def __init__(self, name: str):
+        self.name = name
+        self.start = None
+        self.end = None
+
+    def record_start(self, rank):
+        self.start = torch.cuda.Event(enable_timing=True)
+        with torch.cuda.device(rank):
+            self.start.record()
+
+    def record_end(self, rank):
+        self.end = torch.cuda.Event(enable_timing=True)
+        with torch.cuda.device(rank):
+            self.end.record()
+
+    def elapsed_time(self, rank):
+        self.start.synchronize()
+        self.end.synchronize()
```

Comment:
This means if this is called before the CUDA ops are done, it will be perf impacts. Shall we make the synchronize API explicit? I mean 


```python
    def elapsed_time(self, rank):
        if not self.end.query():
            raise RuntimeError("......")
        return self.start.elapsed_time(self.end)

    def synchronize():
        ....
```

Reply:
okay ill add the synchronize API.